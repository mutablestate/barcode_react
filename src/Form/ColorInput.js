import React from 'react';
import PropTypes from 'prop-types';

const colors = {
  primaryColor: ['red', 'blue', 'yellow'],
  secondaryColor: ['grey', 'purple', 'green']
};

const capitalize = name => name.charAt(0).toUpperCase() + name.slice(1);

// ColorInput
const ColorInput = ({ label, name, color, onChange }) => {
  const options = colors[name].map(color => {
    const label = capitalize(color);
    return (
      <option
        key={color}
        value={color}
        data-test={`ColorInput:option:${color}`}
      >
        {label}
      </option>
    );
  });

  return (
    <div className="form-control">
      <label>{label}</label>
      <select
        name={name}
        value={color}
        onChange={onChange}
        data-test={`ColorInput:select:${name}`}
      >
        {options}
      </select>
    </div>
  );
};

ColorInput.propTypes = {
  /** The input label */
  label: PropTypes.string.isRequired,
  /** The input tag name attribute value */
  name: PropTypes.oneOf(['primaryColor', 'secondaryColor']).isRequired,
  /** The currently selected color value  */
  color: PropTypes.string.isRequired,
  /** The onChange event handler  */
  onChange: PropTypes.func.isRequired
};

export default ColorInput;
