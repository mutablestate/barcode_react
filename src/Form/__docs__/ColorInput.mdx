---
name: ColorInput
route: /form/color-input
menu: Form
---

import { Playground, Props } from 'docz';
import ColorInput from '../ColorInput';

# ColorInput

A basic html select field for colors.

## Props

<Props of={ColorInput} />

## Basic usage

<Playground>
  <form>
    <ColorInput
      label={'Color picker'}
      name={'primaryColor'}
      color={'red'}
      onChange={() => {}}
    />
  </form>
</Playground>
