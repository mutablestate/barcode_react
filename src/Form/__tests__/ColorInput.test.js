import React from 'react';
import renderer from 'react-test-renderer';

import ColorInput from '../ColorInput';

const requiredProps = {
  label: 'LABEL',
  name: 'primaryColor',
  color: 'red',
  onChange: jest.fn()
};

describe('ColorInput', () => {
  it('must render correctly with required props', () => {
    const component = renderer
      .create(<ColorInput {...requiredProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
