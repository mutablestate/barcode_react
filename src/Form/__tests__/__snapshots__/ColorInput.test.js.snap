// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInput must render correctly with required props 1`] = `
<div
  className="form-control"
>
  <label>
    LABEL
  </label>
  <select
    data-test="ColorInput:select:primaryColor"
    name="primaryColor"
    onChange={[MockFunction]}
    value="red"
  >
    <option
      data-test="ColorInput:option:red"
      value="red"
    >
      Red
    </option>
    <option
      data-test="ColorInput:option:blue"
      value="blue"
    >
      Blue
    </option>
    <option
      data-test="ColorInput:option:yellow"
      value="yellow"
    >
      Yellow
    </option>
  </select>
</div>
`;
